examples/message.js

```diff
@@ -0,0 +1,22 @@
+var phantom = require("../lib");
+var _ph, _page, _outObj;
+
+phantom.create().then(ph => {
+    _ph = ph;
+    return _ph.createPage();
+}).then(page => {
+    _page = page;
+    _ph.onMessage(function(resourceUrl) {
+    	console.log('Node context received resourceUrl: ', resourceUrl);
+    });
+    _page.property('onResourceRequested', function(resource, networkRequest, callbackMessagePrefix) {
+    	console.log(callbackMessagePrefix + resource.url);
```

Comment:
This is the part that is really confusing. How would I as a developer know that `console.log` will send a message back to NodeJS. It's a lot of voodoo magic. :( 

I would improve upon this idea by allowing you to have "special" objects like `outObject` where the user doesn't know this is happening.